code/modules/paperwork/paper.dm

```diff
@@ -1,160 +1,189 @@
 
 /obj/item/paper/proc/update_space(var/new_text)
 	if(new_text)
 		free_space -= length(strip_html_properly(new_text))
 
 /obj/item/paper/examine(mob/user, distance)
 	. = ..()
-	if(name != "sheet of paper")
+	if(name != initial(name))
 		to_chat(user, "It's titled '[name]'.")
+
 	if(distance <= 1)
-		show_content(usr)
+		interact(user, readonly = TRUE)
 	else
-		to_chat(user, "<span class='notice'>You have to go closer if you want to read it.</span>")
+		to_chat(user, SPAN_NOTICE("You have to go closer if you want to read it."))
 
-/obj/item/paper/proc/show_content(mob/user, forceshow)
-	var/show_info = user.handle_reading_literacy(user, info, FALSE, (forceshow || get_dist(src, user) <= 1))
+/obj/item/paper/interact(mob/user, var/forceshow = FALSE, var/readonly = FALSE)
+	var/show_info = user.handle_reading_literacy(user, readonly? info : info_links, FALSE, (forceshow || get_dist(src, user) <= 1))
 	if(show_info)
-		show_browser(user, "<HTML><HEAD><TITLE>[name]</TITLE></HEAD><BODY bgcolor='[color]'>[show_info][stamps]</BODY></HTML>", "window=[name]")
+		user.set_machine(src)
+		show_browser(user, "<HTML><HEAD><TITLE>[name]</TITLE></HEAD><BODY bgcolor='[color]'>[show_info][stamp_text]</BODY></HTML>", "window=[name]")
 		onclose(user, "[name]")
-
-/obj/item/paper/verb/rename()
-	set name = "Rename paper"
-	set category = "Object"
-	set src in usr
-
-	if((MUTATION_CLUMSY in usr.mutations) && prob(50))
-		to_chat(usr, "<span class='warning'>You cut yourself on the paper.</span>")
-		return
-	var/n_name = sanitize_safe(input(usr, "What would you like to label the paper?", "Paper Labelling", null)  as text, MAX_NAME_LEN)
-
-	// We check loc one level up, so we can rename in clipboards and such. See also: /obj/item/photo/rename()
-	if(!n_name || !CanInteract(usr, global.deep_inventory_topic_state))
-		return
-	n_name = usr.handle_writing_literacy(usr, n_name)
-	if(n_name)
-		SetName(n_name)
-	add_fingerprint(usr)
+	return TRUE
 
 /obj/item/paper/attack_self(mob/user)
 	if(user.a_intent == I_HURT)
-		if(icon_state == "scrap")
-			user.show_message("<span class='warning'>\The [src] is already crumpled.</span>")
+		if(is_crumpled)
+			user.show_message(SPAN_WARNING("\The [src] is already crumpled."))
 			return
 		//crumple dat paper
-		info = stars(info,85)
+		crumple()
 		user.visible_message("\The [user] crumples \the [src] into a ball!")
-		icon_state = "scrap"
-		return
-	user.examinate(src)
+		return TRUE
+
+	interact(user, readonly = FALSE) //Allow us writing on paper since we're holding it somwhere
+
 	if(rigged && (global.current_holiday?.name == "April Fool's Day"))
-		if(spam_flag == 0)
-			spam_flag = 1
-			playsound(loc, 'sound/items/bikehorn.ogg', 50, 1)
+		if(!is_honking)
+			is_honking = TRUE
+			playsound(loc, 'sound/items/bikehorn.ogg', 50, TRUE)
 			spawn(20)
-				spam_flag = 0
+				is_honking = FALSE
+	return TRUE
 
 /obj/item/paper/attack_ai(mob/living/silicon/ai/user)
-	show_content(user)
+	interact(user)
```

Comment:
Everyone else is read-only but not us?

Reply:
I'm just confused as to why it's not read only for AI: presumably it shouldn't be modifying the paper?

Reply:
Hmm, yeah, that's true. I always forgot that there used to be a use and a show_content proc.